src/core/util/byte_stream.cc

```diff
@@ -100,6 +100,8 @@ void OutputByteStream::ProduceData(std::size_t amount) {
 }
 
 void OutputByteStream::WriteData(const std::uint8_t* data, std::size_t len) {
+  if (!len)
+    return;
```

Comment:
>But I would like to write things like
> packet->SetIPAddress(nullptr, 0)

I'm not 100% sure why'd you want to do this or why you *don't* want to throw on 0 length or `nullptr`. If we must do this, I think we should add debug logging for the odd times that we *would* want to do this. What do you think?

Also, while we're here: `if (!data || !len)`.